Frontend/frontend/src/Components/Navigation.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { Link, Redirect } from 'react-router-dom';
+
+import './Navigation.css';
+import User from '../Util/User';
+
+export default class Navigation extends React.Component {
+  render() {
+    return <nav className="menu">
+      { !User.getUserState() ? <Redirect to="/login/"/> : ""}
```

Comment:
I think a comment here could be valuable, since there's negation and a ternary operator it's a little confusing to read

Reply:
Good catch, actually I have moved this to its own component "RequireLogin", which is required for all /app/ urls